osmaxx/core/templatetags/navigation.py

```diff
@@ -1,19 +1,34 @@
-# Adopted from https://djangosnippets.org/snippets/2783/
+from urllib.parse import urlsplit, urlunsplit
+
+import logging
 from django import template
 from django.conf import settings
+from django.http import HttpRequest
+
+logger = logging.getLogger(__name__)
 
 register = template.Library()
 
 
 @register.filter
-def siteabsoluteurl(absoluteurl, request):
+def siteabsoluteurl(relative_or_absolute_url: str, request: HttpRequest) -> str:
     """
-    receives an argument in the form of an absolute url <absoluteurl>
-    Returns the same url with all the necessary path information to be used offsite
+    Produce URL suitable for offsite use.
+
+    Args:
+        relative_or_absolute_url: A relative, scheme-relative or absolute URL
+        request: A Django request object, from which to take the missing parts
+
+    Returns:
+        An absolute URL, incl. scheme, host, etc.
     """
-    absolute_url = request.build_absolute_uri(absoluteurl)
-    if settings.OSMAXX.get('SECURED_PROXY', False) and not absolute_url.startswith('https'):
-        if absolute_url.startswith('http'):
-            absolute_url = absolute_url[4:]
-        absolute_url = 'https' + absolute_url
+    absolute_url = request.build_absolute_uri(relative_or_absolute_url)
+    if settings.OSMAXX.get('SECURED_PROXY', False):
+        url_components = urlsplit(absolute_url)
```

Comment:
Yeah. IIRC, we already considered using `urllib.parse` back when this filter was first implemented, but didn't use it because we didn't want to "introduce unnecessary dependencies". :thinking: Noticing recently that `request.build_absolute_uri()` (which we already used then) uses `urllib.parse`, too, I thought that point was moot. :stuck_out_tongue_closed_eyes: